src/prefect/tasks/gcp/bigquery.py

```diff
@@ -403,6 +404,153 @@ def run(
         result = load_job.result()  # block until job is finished
 
 
+class BigQueryLoadFile(Task):
+    """
+    Task for insert records in a Google BigQuery table via a [load job](https://cloud.google.com/bigquery/docs/loading-data).
+    Note that all of these settings can optionally be provided or overwritten at runtime.
+
+    Args:
+        - file (Union[str, path-like object], optional): A string or path-like object of the file to be loaded
+        - rewind (bool, optional): if True, seek to the beginning of the file handle before reading the file
+        - size (int, optional):  the number of bytes to read from the file handle. If size is None or large,
+            resumable upload will be used. Otherwise, multipart upload will be used.
+        - dataset_id (str, optional): the id of a destination dataset to write the
+            records to
+        - table (str, optional): the name of a destination table to write the
+            records to
+        - project (str, optional): the project to initialize the BigQuery Client with; if not provided,
+            will default to the one inferred from your credentials
+        - schema (List[bigquery.SchemaField], optional): the schema to use when creating the table
+        - location (str, optional): location of the dataset that will be queried; defaults to "US"
+        - credentials_secret (str, optional, DEPRECATED): the name of the Prefect Secret
```

Comment:
Left over credentials secret in this task 😄

Reply:
@joshmeek : had a look, to fix it we need to remove all `credentials_secret` in `tasks/gcp/bigquery.py` since e.g. `TestBigQueryCredentialsandProjects` also still uses these.

Reply:
@joshmeek happy to remove all `credentials_secret` in `bigquery.py`, but that would be a breaking change. Just let me know what I can do to help.

Reply:
Good call! There's a large refactor coming in the `0.11.0` release which deals with stuff like these `credentials_secrets` to fall in line with Prefect's new preferred way of using sensitive information (#2326). I say let's remove the use of all `credentials_secret` options in your new tasks and then once `0.11.0` is released (sometime later this week or early next week) this will be in line with the new authentication patterns and we can merge 😄